### PR TITLE
Added test to listunspent RPC

### PR DESCRIPTION
**Pull request by joaopaulofonseca**
Added test to the listunspent call. This RPC had no individual test yet.

cc @promag

### CODE REVIEW

File: qa/rpc-tests/listunspent.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import *
+
+class ListUnspentTest(BitcoinTestFramework):
+
+    def setup_chain(self):
+        print("Initializing test directory "+self.options.tmpdir)
+        initialize_chain_clean(self.options.tmpdir, 2)
```
Comment (pedrobranco) [inline qa/rpc-tests/listunspent.py]: Initializing 2 nodes when bellow you use only `self.nodes[0]`?

File: qa/rpc-tests/listunspent.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+    def setup_network(self, split=False):
+        self.nodes = start_nodes(2, self.options.tmpdir)
+
+        connect_nodes_bi(self.nodes,0,1)
```
Comment (pedrobranco) [inline qa/rpc-tests/listunspent.py]: No need to link the 2 nodes.

File: qa/rpc-tests/listunspent.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+        self.is_network_split=False
+        self.sync_all()
```
Comment (pedrobranco) [inline qa/rpc-tests/listunspent.py]: same as before.

File: qa/rpc-tests/listunspent.py [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+    def run_test(self):
+
+        # check there's no unspent output availabe on any nodes
+        assert_equal(len(self.nodes[0].listunspent()), 0)
+        assert_equal(len(self.nodes[1].listunspent()), 0)
+
+        # mine enough blocks to generate one UTXO
+        self.nodes[0].generate(101)
+        self.sync_all()
```
Comment (pedrobranco) [inline qa/rpc-tests/listunspent.py]: same as before.

File: qa/rpc-tests/listunspent.py [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+        # sends some coins between nodes, to spend some available unspent outputs
+        txid = self.nodes[0].sendtoaddress(self.nodes[1].getnewaddress(), 1)
+
+        # since the only available UTXO was spent, there sould be none after the transaction
+        assert_equal(len(self.nodes[0].listunspent()), 0)
+
+        # mine one more block to generate some more UTXOs
+        self.nodes[0].generate(1)
+        self.sync_all()
```
Comment (pedrobranco) [inline qa/rpc-tests/listunspent.py]: same as before.

**Comment by promag** [comment]
Don't forget to add some imports and sync between nodes to test if unspents are _catched_.

**Comment by joaopaulofonseca** [comment]
@promag @pedrobranco yep. I'm improving this test :)

**Comment by jpdffonseca** [comment]
Deprecated by bitcoin PR https://github.com/bitcoin/bitcoin/pull/7822.

**Pull request closed without merging**